src/Diffusion.jl

```diff
@@ -108,4 +112,94 @@ function nmolabsorb(σ, steps, pos, r, molecules) #simulates multiple molecules
     return molhistories
 end
 
+### Analysis ###
+ 
+function molremaining(σ, steps, pos, r, molecules, boundary)
+    if boundary == "absorb"
+        trial = nmolabsorb(σ, steps, pos, r, molecules)
+    elseif boundary == "reflect"
+        trial = nmolreflect(σ, steps, pos, r, molecules)
+    else
+        throw(DomainError("Please specify boudary condition as 'absorb' or 'reflect'"))
+    end
+    lengths = molpathlengths(trial)
+    return percentremaining(lengths)
+    end
+    
+    function molpathlengths(trial) #input the molhistories given my nmolreflect or nmolabsorb
+    lengthsofmolpaths = []         #outputs a vector showing the lengths of each molecules' path
```

Comment:
```suggestion
    lengthsofmolpaths = Int[]         #outputs a vector showing the lengths of each molecules' path
```

Reply:
This is important for getting good performance if you do any analysis on this vector. When the elements are of type `Any`, Julia can't optimize the code very well.